examples/eap_server.rs

```diff
@@ -0,0 +1,169 @@
+//                         (MyID) ->
+//                                                <- RADIUS
+//                                                Access-Challenge/
+//                                                EAP-Message/EAP-Request
+//                                                OTP/OTP Challenge
+//                         <- EAP-Request/
+//                         OTP/OTP Challenge
+// EAP-Response/
+// OTP, OTPpw ->
+//                         RADIUS Access-Request/
+//                         EAP-Message/EAP-Response/
+//                         OTP, OTPpw ->
+//                                                 <- RADIUS
+//                                                 Access-Accept/
+//                                                 EAP-Message/EAP-Success
+//                                                 (other attributes)
+//                         <- EAP-Success
+    pub async fn handle_eap_identity(&self, req_packet: &Packet, conn: &UdpSocket, req: &Request, ieap: &EAP ) -> Result<usize, io::Error> {
+        let mut eap = EAP::new(); //ieap.clone();
+        let mut ac_packet = req_packet.make_response_packet(Code::AccessChallenge);
+        eap.code = EAPCode::Response;
+        eap.typ  = EAPType::MD5Challenge;
+        eap.id   = ieap.id;
+        eap.len  = eap.recalc_len();
+        println!("Response EAP: {}", &eap);
+        rfc2869::add_eap_message(&mut ac_packet, &eap.to_bytes()[..]);
+        let ac_bytes = &ac_packet.encode().unwrap();
+        let authenticator = Packet::create_respose_authenticator(&ac_bytes[..], &req_packet.encode().unwrap()[..], &req_packet.get_secret());
+        rfc2869::add_message_authenticator(&mut ac_packet, &authenticator);
+        let result = conn.send_to(&ac_packet.encode().unwrap(), req.get_remote_addr()).await.unwrap();
+        Ok(result)
+    }
+}
+
+#[async_trait]
+impl RequestHandler<(), io::Error> for MyRequestHandler {
+    async fn handle_radius_request(
+        &self,
+        conn: &UdpSocket,
+        req: &Request,
+    ) -> Result<(), io::Error> {
+        let req_packet = req.get_packet();
+        // println!("Req:\n{:#?}", req_packet.clone());
+        let maybe_user_name_attr = rfc2865::lookup_user_name(req_packet);
+        let maybe_user_password_attr = rfc2865::lookup_user_password(req_packet);
+        info!("maybe usr pass looked up");
+        let maybe_eap_message = rfc2869::lookup_eap_message(req_packet);
+        info!("maybe eap looked up");
+        match maybe_eap_message {
+            Some(e) => {
+                let ieap = EAP::from_bytes(&e);
+                println!("EAP Message: {}", &ieap);
+                let result = self.handle_eap_identity(req_packet, conn, req, &ieap).await.unwrap();
+                println!("Sent challenge-response {} bytes", &result);
+                ieap
+            },
+            None => {
+                println!("No eap message found");
+                EAP::new()
+            }
+        };
```

Comment:
It seems the result of this `match` expression is discarded. What does this code intend?

Reply:
It is, but it satisfies the constraints of the match (maybe thats not needed anymore, been doing it this way for a few years).